components/core/src/fs.rs

```diff
@@ -201,13 +201,39 @@ pub fn find_command(command: &str) -> Option<PathBuf> {
                 if candidate.is_file() {
                     return Some(candidate);
                 }
+                else {
+                    match find_command_with_pathext(&candidate) {
+                        Some(result) => return Some(result),
+                        None => {}
+                    } 
+                }
             }
             None
         }
         None => None,
     }
 }
 
+// Windows relies on path extensions to resolve commands like `docker` to `docker.exe`
+// Path extensions are found in the PATHEXT environment variable.
+fn find_command_with_pathext(candidate: &PathBuf) -> Option<PathBuf> {
+    match henv::var_os("PATHEXT") {
+        Some(pathexts) => {
+            for pathext in env::split_paths(&pathexts) {
+                let mut source_candidate = candidate.to_path_buf();
+                let extension = pathext.to_str().unwrap().trim_matches('.');
+                source_candidate.set_extension(extension);
+                let current_candidate = source_candidate.to_path_buf(); 
+                if current_candidate.is_file() {
+                    return Some(current_candidate);
+                }
+            }
```

Comment:
Like in Ruby, the last line of a function is the return value in Rust. Short-circuiting the function with `return Some(T)` on 228 is great, but we should fall through to a None below this line. That would allow you to remove the `;` on 233 & the explicit call to `return None` on 232.

These are small suggestions but they'll make the function control flow read a bit easier and be more "rust-like"